src/test/java/io/tiledb/spark/TimeTravellingTest.java

```diff
@@ -0,0 +1,246 @@
+package io.tiledb.spark;
+
+import static io.tiledb.java.api.ArrayType.TILEDB_DENSE;
+import static io.tiledb.java.api.Layout.TILEDB_ROW_MAJOR;
+import static io.tiledb.java.api.QueryType.TILEDB_WRITE;
+
+import io.tiledb.java.api.*;
+import java.math.BigInteger;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import org.apache.spark.sql.*;
+import org.junit.*;
+
+public class TimeTravellingTest extends SharedJavaSparkSession {
+
+  private static Context ctx;
+  private String arrayURI;
+
+  @Before
+  public void setup() throws Exception {
+    ctx = new Context();
+    arrayURI = "time_travelling_test_array";
+    deleteArrays();
+  }
+
+  @After
+  public void delete() throws Exception {
+    deleteArrays();
+    ctx.close();
+  }
+
+  private void deleteArrays() throws TileDBError {
+    if (Files.exists(Paths.get(arrayURI))) {
+      TileDBObject.remove(ctx, arrayURI);
+    }
+  }
+
+  @Test
+  public void testTimeTravelling() throws Exception {
+    // create array
+    arrayCreate();
+    // updates
+    arrayWrite1();
```

Comment:
Do you mind to make the timestamp a parameter to these writer functions, and add a comment that we are doing the writes at specific times?

Reply:
Improved this.